src/NHibernate/IQuery.cs

```diff
@@ -119,6 +123,35 @@ public partial interface IQuery
 		/// <returns></returns>
 		IEnumerable<T> Enumerable<T>();
 
+		/// <summary>
+		/// Return the query results as an <see cref="IEnumerable"/>. If the query contains multiple results
+		/// per row, the results are returned in an instance of <c>object[]</c>.
+		/// </summary>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		/// <remarks>
+		/// <p>
+		/// Entities returned as results are initialized on demand. The first SQL query returns
+		/// identifiers only.
+		/// </p>
+		/// <p>
+		/// This is a good strategy to use if you expect a high number of the objects
+		/// returned to be already loaded in the <see cref="ISession"/> or in the 2nd level cache.
+		/// </p>
+		/// </remarks>
+		// Since v5.3
+		[Obsolete("This method has no more usages and will be removed in a future version")]
+		Task<IEnumerable> EnumerableAsync(CancellationToken cancellationToken = default(CancellationToken));
+
+		/// <summary>
+		/// Strongly-typed version of <see cref="Enumerable()"/>.
+		/// </summary>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		/// <typeparam name="T"></typeparam>
+		/// <returns></returns>
+		// Since v5.3
+		[Obsolete("This method has no more usages and will be removed in a future version")]
+		Task<IEnumerable<T>> EnumerableAsync<T>(CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Shouldn't we suggest what should be used instead?

Reply:
How about adding support for conversion rule `Obsolete` in AsyncGenerator.yml to avoid adding manually those methods.
Something like: 
```yml
#Since v5.3
    - conversion: Obsolete
      name: EnumerableAsync
      description: "Use AsyncEnumerable instead"
```

Reply:
> Shouldn't we suggest what should be used instead?

Description updated.

> How about adding support for conversion rule Obsolete

Is seems a good idea, I will try to implement it in the near future.